src/isomedia/drm_sample.c

```diff
@@ -967,14 +973,32 @@ void gf_isom_cenc_merge_saiz_saio(GF_SampleEncryptionBox *senc, GF_SampleTableBo
 		senc->cenc_saiz->sample_count++;
 	}
 
+  // BOOM
 	if (!senc->cenc_saio->entry_count) {
-		senc->cenc_saio->offsets_large = (u64 *)gf_malloc(sizeof(u64));
-		senc->cenc_saio->offsets_large[0] = offset;
-		senc->cenc_saio->entry_count ++;
+    if (offset>0xFFFFFFFFULL) {
+      senc->cenc_saio->version = 1;
+      senc->cenc_saio->offsets_large = (u64 *)gf_malloc(sizeof(u64));
+      senc->cenc_saio->offsets_large[0] = offset;
+      senc->cenc_saio->entry_count ++;
+    } else {
+      senc->cenc_saio->version = 0;
+  		senc->cenc_saio->offsets = (u32 *)gf_malloc(sizeof(u32));
+  		senc->cenc_saio->offsets[0] = offset;
+  		senc->cenc_saio->entry_count ++;
+    }
 	} else {
-		senc->cenc_saio->offsets_large = (u64*)gf_realloc(senc->cenc_saio->offsets_large, sizeof(u64)*(senc->cenc_saio->entry_count+1));
-		senc->cenc_saio->offsets_large[senc->cenc_saio->entry_count] = offset;
-		senc->cenc_saio->entry_count++;
+    if (offset>0xFFFFFFFFULL) {
+      GF_LOG(GF_LOG_ERROR, GF_LOG_CONTAINER, ("[iso file] BOOM setting 64\n"));
+      senc->cenc_saio->version = 1;
+      senc->cenc_saio->offsets_large = (u64*)gf_realloc(senc->cenc_saio->offsets_large, sizeof(u64)*(senc->cenc_saio->entry_count+1));
+  		senc->cenc_saio->offsets_large[senc->cenc_saio->entry_count] = offset;
```

Comment:
if you realloc to 64 bit the 32 bit offets (ie saio version change) you will loose the offsets, you need to copy them manually. Apart from that and cleanup of commented test code we should be good to merge no?